### PR TITLE
[Usa1979P1] Add proof

### PR DESCRIPTION
**Pull request by casavaca**

### CODE REVIEW

File: Compfiles/Usa1979P1.lean [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+apart from permutations, of the Diophantine Equation $n₁⁴ + n₂⁴ + ... + n₁₄⁴ = 1599.
+-/
+
+namespace Usa1979P1
+
+structure Perm14 where
+  s : Multiset ℕ
+  p : Multiset.card s = 14
+
+set_option diagnostics true in
```
Comment (dwrensha) [inline Compfiles/Usa1979P1.lean:26]: Why is is `set_option` here? Please delete it. Or, if there is a good reason for it, add a comment explaining.

File: Compfiles/Usa1979P1.lean [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import ProblemExtraction
+
+problem_file { tags := [.Algebra, .Inequality] }
+
+/-!
+# USA Mathematical Olympiad 1979, Problem 1
+
+Determine all non-negative integral solutions (n₁, n₂, ..., n₁₄) if any,
+apart from permutations, of the Diophantine Equation $n₁⁴ + n₂⁴ + ... + n₁₄⁴ = 1599.
```
Comment (dwrensha) [inline Compfiles/Usa1979P1.lean:17]: Unclosed `$`. Please either delete it, or do everything in LaTeX style.

File: Compfiles/Usa1979P1.lean [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+-/
+
+namespace Usa1979P1
+
+structure Perm14 where
```
Comment (dwrensha) [inline Compfiles/Usa1979P1.lean:22]: I don't understand what's going on here. This does not look like a permutation to me.

File: Compfiles/Usa1979P1.lean (hunk 1 above)
Comment (casavaca) [reply Compfiles/Usa1979P1.lean:26]: debug code accidentally left. Now deleted.

File: Compfiles/Usa1979P1.lean (hunk 2 above)
Comment (casavaca) [reply Compfiles/Usa1979P1.lean]: Changed to LaTeX style.

File: Compfiles/Usa1979P1.lean (hunk 3 above)
Comment (casavaca) [reply Compfiles/Usa1979P1.lean]: Renamed to `MultisetNatOfLen14` representing $(n_1,n_2,\dots , n_{14})$ (duplication allowed).

**Comment by dwrensha** [comment]
> I used `Equiv.Perm` and `Fin 14 -> Nat` to remove the need for `MultisetNatOfLen14` in this commit: [46cebda](https://github.com/dwrensha/compfiles/commit/46cebda929e0862e2420856436eadfcf6d8f8089)

Reverted in 46cebda929e0862e2420856436eadfcf6d8f8089. 🤦 That made the problem statement not quite right.

**Pull request merged**